libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
+    for (BI_NetLine* nl : currentAnchor.get().getNetLines()) {
+      // Don't visit a segment twice
+      if (visitedSegments.contains(nl->getUuid())) {
+        continue;
+      }
+      // Only visit selected segments
+      if (nl->isSelected()) {
+        if (nextNetline.has_value()) {
+          // There's already another connected and selected netline
+          throw LogicError(__FILE__, __LINE__,
+                           tr("Selected segment may not have branches"));
+        }
+
+        totalLength += nl->getLength();
+        nextNetline = *nl;
+        visitedSegments.insert(nl->getUuid());
+      }
+    }
+    if (nextNetline.has_value()) {
+      currentNetline = std::cref(*nextNetline);
+      currentAnchor =
+          std::cref(*nextNetline->getOtherPoint(currentAnchor.get()));
+    } else {
+      break;
+    }
+  }
+}
+
+bool BES_Select::measureSelectedItems(BI_NetLine& netline) noexcept {
```

Comment:
Probably the netline could be `const`(?):

```suggestion
bool BES_Select::measureSelectedItems(const BI_NetLine& netline) noexcept {
```